src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,304 @@
+import React from 'react';
+import { useIntl } from 'react-intl';
+import { get, filter } from 'lodash-es';
+
+import { FormattedMessage } from 'react-intl';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+import TextInput from '../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../components/DeleteButton';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+
+const intl = useIntl();
+
+function recomposeAllRelationshipObjectFromModifiedCategoryObject(
+  currentAllRelationshipObject,
+  modifiedTargetCategoryObj,
+) {
+  const nonTargeCategoryObjects = filter(
+    currentAllRelationshipObject,
+    currentRelationshipCategory =>
+      currentRelationshipCategory.category !==
+      modifiedTargetCategoryObj?.category,
+  );
+  return [...nonTargeCategoryObjects, modifiedTargetCategoryObj];
+}
+
+function deleteCategory(allRelationshipsObj, category) {
+  return allRelationshipsObj.filter(
+    singleRelationshipObj =>
+      singleRelationshipObj?.category !== category,
+  );
+}
+
+function deleteRelationship(
+  allRelationshipsObj,
+  currentCategory,
+  relationship,
+) {
+  const targetRelationshipCategoryObjArr = filter(
+    allRelationshipsObj,
+    currentRelationshipCategory =>
+      currentRelationshipCategory.category === currentCategory,
+  );
+  // if (targetRelationshipCategoryObjArr?.length > 1) return; // TODO how do we handle duplicate relationship categories/types?
```

Comment:
Post MVP? Just treat them all as independent, even if they're the same exact? Prevent users from creating categories or relationships that already exist?

Reply:
well since the categories are keys for an object it shouldn't be possible to duplicate them. but we should make sure things aren't buggy if they try. 

different story for the relationships since they are just an array. i do think they should be prohibited from duplicating them.

Reply:
Done. Handled the relationship duplication in the actual button submission since it creates undesired errors while the user is making progress on typing the new potential relationship.